src/Instrumentation/Laravel/src/HttpInstrumentation.php

```diff
@@ -80,15 +79,11 @@ public static function register(CachedInstrumentation $instrumentation): void
                     $span->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $response->getProtocolVersion());
                     $span->setAttribute(TraceAttributes::HTTP_RESPONSE_BODY_SIZE, $response->headers->get('Content-Length'));
                 }
-                if(($route = Route::getCurrentRoute()?->uri()) !== null) {
-                    $request = ($params[0] instanceof Request) ? $params[0] : null;
-
-                    if (! str_starts_with($route, '/')) {
-                        $route = '/' . $route;
-                    }
 
+                $request = ($params[0] instanceof Request) ? $params[0] : null;
+                if (($route = $request?->route()?->uri()) !== null) {
```

Comment:
```
Error: src/HttpInstrumentation.php:84:52: PossiblyInvalidMethodCall: Cannot call method on possible string variable $__tmp_nullsafe__4512 (see https://psalm.dev/113)
```

Reply:
https://laravel.com/api/6.x/Illuminate/Http/Request.html#method_route

So `Request::route()` returns `Route|object|string|null`. The implementation here assumes it's always a route. Can you account for object & string as well?

Reply:
I think a viable solution could be as i proposed here https://github.com/open-telemetry/opentelemetry-php-contrib/pull/219/files/41e7dac805c26e2ea179a6145697545f87d5acc7#r1411897050

Reply:
I've stepped through this now. Once the `Illuminate\Routing\Router` dispatches the request, the route resolver is set as a closure which will only ever return a `Route` object.

This feels cleaner than using `$request->route()` personally, which takes additional parameters and code-paths which the tooling flags as having multiple return types.